apps/openmw/mwrender/npcanimation.cpp

```diff
@@ -893,6 +904,101 @@ void NpcAnimation::addControllers()
+                switch(weaponType)
+                {
+                    case ESM::Weapon::ShortBladeOneHand:
+                    case ESM::Weapon::LongBladeOneHand:
+                    case ESM::Weapon::BluntOneHand:
+                    case ESM::Weapon::AxeOneHand:
+                        weaptype = ESM::PRT_WeaponOneHand;
+                        break;
+                    case ESM::Weapon::LongBladeTwoHand:
+                    case ESM::Weapon::BluntTwoClose:
+                    case ESM::Weapon::AxeTwoHand:
+                        weaptype = ESM::PRT_WeaponTwoClose;
+                        break;
+                    case ESM::Weapon::BluntTwoWide:
+                    case ESM::Weapon::SpearTwoWide:
+                        weaptype = ESM::PRT_WeaponTwoWide;
+                        break;
+                    case ESM::Weapon::MarksmanBow:
+                        weaptype = ESM::PRT_WeaponBow;
+                        break;
+                    case ESM::Weapon::MarksmanCrossbow:
+                        weaptype = ESM::PRT_WeaponCrossbow;
+                        break;
+                    default:
+                        break;
+                }
+            }
+
+            if (weaptype != ESM::PRT_Count)
+                addOrReplaceIndividualPart(weaptype, -1, 1, mesh, !weapon->getClass().getEnchantment(*weapon).empty(), &glowColor);
+        }
+    }
+
+    MWWorld::ConstContainerStoreIterator ammo = inv.getSlot(MWWorld::InventoryStore::Slot_Ammunition);
+    if (ammo != inv.end() && ammo.getType() == MWWorld::ContainerStore::Type_Weapon)
+    {
+        if (weapon == inv.end()) return;
+
+        const std::string &type = weapon->getClass().getTypeName();
+
+        // Show quiver only if the ranged weapon with the suitable ammo is equipped.
+        bool needToShowQuiver = false;
+        if(type == typeid(ESM::Weapon).name())
+        {
+            const MWWorld::LiveCellRef<ESM::Weapon> *ref = weapon->get<ESM::Weapon>();
+            ESM::Weapon::Type weaponType = (ESM::Weapon::Type)ref->mBase->mData.mType;
+
+            const MWWorld::LiveCellRef<ESM::Weapon> *ammoRef = ammo->get<ESM::Weapon>();
+            ESM::Weapon::Type ammoType = (ESM::Weapon::Type)ammoRef->mBase->mData.mType;
+
+            if (weaponType == ESM::Weapon::MarksmanBow && ammoType == ESM::Weapon::Arrow)
+                needToShowQuiver = true;
+            else if (weaponType == ESM::Weapon::MarksmanCrossbow && ammoType == ESM::Weapon::Bolt)
+                needToShowQuiver = true;
+        }
+
+        if (needToShowQuiver)
+        {
+            osg::Vec4f glowColor = getEnchantmentColor(*ammo);
+            std::string ammoMesh = ammo->getClass().getModel(*ammo);
+            if (ammoMesh.empty()) return;
+
+            std::string quiverMesh;
+            quiverMesh.append(ammoMesh.begin(),ammoMesh.end()-4);
+            quiverMesh.append("_quiver.nif");
```

Comment:
What about osgt, the native mesh format™®©?

Reply:
> "Just a proof of concept for now."

I used a stopgap solution to show a possibility to implement such feature in OpenMW.
I guess there should be better possibilities than using suffix.

Reply:
I wonder how quivers are implemented in other games (e.g. in Skyrim).